src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/contactperson/ContactPersonService.kt

```diff
@@ -427,6 +427,32 @@ class ContactPersonService(
       telemetryClient.trackEvent("$entityName-update-ignored", telemetryMap)
     }
   }
+  suspend fun contactPhoneDeleted(event: ContactPhoneDeletedEvent) {
+    val entityName = CONTACT_PHONE.entityName
+    val dpsContactPhoneId = event.additionalInformation.contactPhoneId
+    val dpsContactId = event.contactId()
+    val telemetryMap = mutableMapOf(
+      "dpsContactPhoneId" to dpsContactPhoneId.toString(),
+      "dpsContactId" to dpsContactId.toString(),
+      "nomisPersonId" to dpsContactId.toString(),
+    )
+
+    if (event.didOriginateInDPS()) {
+      mappingApiService.getByDpsContactPhoneIdOrNull(dpsContactPhoneId)?.nomisId?.also {
+        telemetryMap["nomisPhoneId"] = it.toString()
+        nomisApiService.deletePersonPhone(
+          personId = dpsContactId,
+          phoneId = it,
+        )
+        // TODO - Delete mapping - though behaviour wise all will work since the "from nomis" service will delete anyway
```

Comment:
I will add this in later, but it is not 100% necessary since the migration service process *all* deletes regardless of original (so it can handle cascading deletes) - so the mapping will be cleaned up anyway